packages/server/src/controllers/user.ts

```diff
@@ -0,0 +1,170 @@
+//  Required Libraries
+import jwt from 'jsonwebtoken';
+import axios, { AxiosResponse } from 'axios';
+import * as nodemailer from 'nodemailer';
+import * as crypto from 'crypto';
+import { Request, Response } from 'express';
+//  DB Model
+import User from '../models/user.js';
+
+//  Register User
+export const registerUser = async (req: Request, res: Response) => {
+  try {
+    const { email } = req.body;
+    const existingUser = await User.findOne({ email });
+    if (existingUser) { return res.status(400).send('User already Exists!'); }
+    // Generate a random 6-digit OTP
+    const otp = crypto.randomBytes(4).toString('hex');
+    // Save OTP in a new collection. To be implemented
+    const user = new User({
+      email,
+      password: null,
+      files: [],
+      otp,
+      isVerified: false,
+      secret: null,
+    });
+    await user.save();
+
+    const dataToken = jwt.sign({ userID: email, isValid: false, role: 'user' }, process.env.JWT_SECRET as string);
+
+    // Send the OTP to the user's email
+    const transport = nodemailer.createTransport({
+      service: 'Gmail',
+      auth: {
+        // Make sure you add a mail and password
+        user: process.env.EMAIL,
+        pass: process.env.EMAIL_PASS,
+      },
+    });
+
+    // Body of the mail and other information
+    const mailOptions = {
+      from: process.env.EMAIL,
+      to: email,
+      subject: 'Your One time password is here! ',
+      // text: 'Hello ',
+      html: `<b>Hello User your OTP is: ${otp}</b>`,
+    };
+    try {
+      const info = await transport.sendMail(mailOptions);
+      console.log(`Email sent: ${(<any>info).status}`);
+      return res.status(250).json({ message: 'Verification email sent! Please check your inbox.', token: dataToken });
+    } catch (err) {
+      console.log(err);
+      return res.status(500).send('Error sending verification email. Please try again in sometime.');
+    }
+  } catch (err) {
+    console.log(err);
+    res.status(400).send('Error occurred check code');
```

Comment:
What does `check code` means?

Reply:
During development I added certain comments for me to understand and debug. Will update it.